lighthouse-core/lib/traces/metrics-evts.js

```diff
@@ -0,0 +1,204 @@
+   */
+  static get metricsDefinitions() {
+    return [
+      {
+        name: 'Navigation Start',
+        id: 'navstart',
+        getTs: auditResults => {
+          const fmpExt = auditResults['first-meaningful-paint'].extendedInfo;
+          return fmpExt.value.timestamps.navStart;
+        }
+      },
+      {
+        name: 'First Contentful Paint',
+        id: 'ttfcp',
+        getTs: auditResults => {
+          const fmpExt = auditResults['first-meaningful-paint'].extendedInfo;
+          return fmpExt.value.timestamps.fCP;
+        }
+      },
+      {
+        name: 'First Meaningful Paint',
+        id: 'ttfmp',
+        getTs: auditResults => {
+          const fmpExt = auditResults['first-meaningful-paint'].extendedInfo;
+          return fmpExt.value.timestamps.fMP;
+        }
+      },
+      {
+        name: 'Perceptual Speed Index',
+        id: 'psi',
+        getTs: auditResults => {
+          const siExt = auditResults['speed-index-metric'].extendedInfo;
+          return siExt.value.timestamps.perceptualSpeedIndex;
+        }
+      },
+      {
+        name: 'First Visual Change',
+        id: 'fv',
+        getTs: auditResults => {
+          const siExt = auditResults['speed-index-metric'].extendedInfo;
+          return siExt.value.timestamps.firstVisualChange;
+        }
+      },
+      {
+        name: 'Visually Complete 85%',
+        id: 'vc85',
+        getTs: auditResults => {
+          const siExt = auditResults['time-to-interactive'].extendedInfo;
+          return siExt.value.timestamps.visuallyReady;
+        }
+      },
+      {
+        name: 'Visually Complete 100%',
+        id: 'vc',
+        getTs: auditResults => {
+          const siExt = auditResults['speed-index-metric'].extendedInfo;
+          return siExt.value.timestamps.visuallyComplete;
+        }
+      },
+      {
+        name: 'Time to Interactive',
+        id: 'tti',
+        getTs: auditResults => {
+          const ttiExt = auditResults['time-to-interactive'].extendedInfo;
+          return ttiExt.value.timestamps.timeToInteractive;
+        }
+      }
+    ];
+  }
+
+  /**
+   * Returns simplified representation of all metrics' timestamps from monotonic clock
+   * @param {!Array<!AuditResult>} auditResults
+   * @return {!Array<{ts: number, id: string, name: string}>} metrics to consider
+   */
+  gatherMetrics(auditResults) {
+    const metricDfns = Metrics.metricsDefinitions;
+    const resolvedMetrics = [];
+    metricDfns.forEach(metric => {
+      // try/catch in case auditResults is missing an audit result
+      try {
+        resolvedMetrics.push({
+          id: metric.id,
+          name: metric.name,
+          ts: metric.getTs(auditResults)
+        });
+      } catch (e) {
+        log.error('metrics-evts', `${metric.name} timestamp not found: ${e.message}`);
+      }
+    });
+    return resolvedMetrics;
+  }
+
+  /**
+   * Getter for our navigationStart trace event
+   */
+  get navigationStartEvt() {
```

Comment:
make just `getNavigationStartEvt()` or something instead of a getter? Kind of confusing below when this property appears.

Could also be an internal function that the constructor calls, populating the `this.navigationStartEvt` property so the behavior doesn't change from this